### PR TITLE
function drill

### PR DESCRIPTION
**Pull request by ngray122**
Function drill

### CODE REVIEW

File: students/nicole/lab/drills/functions.js [hunk 1]
```diff
@@ -68,11 +68,27 @@
 // ### Infer how many arguments you'll need to pass in 
 
 // Create a function that returns the sum of three numbers
+function NumSum(numOne, numTwo, numThree){
+  return numOne + numTwo + numThree  
+};
+NumSum(4, 4, 4);
```
Comment (ngray122) [inline students/nicole/lab/drills/functions.js]: Am I creating a new variable when I invoke the function?

Comment (jaygraygray) [reply students/nicole/lab/drills/functions.js]: Great question! No, you're not creating a variable when you invoke the function here.

However, if you do this:
`const sum = NumSum(4, 4, 4)`

then `sum` will be the value of the `numOne + numTwo + numThree` operation that's performed in the `NumSum` function. This is the power of the `return` keyword. 

You can perform a bunch of computations or do a bunch of stuff inside of a function, and once all the computing is done, store or return that final value into a variable to be used by the rest of your code.

**Comment by jaygraygray** [comment]
Your commit habits are looking great 🎉

File: students/nicole/lab/drills/functions.js [hunk 2]
```diff
@@ -69,17 +69,62 @@
...
+};
+
+numDiv(24,2);
+
 // Create a function that returns the data type of a variable
 // ## HINT: use the typeof keyword
+let myAge = 34;
+
+function retVar(paramOne){
+   return typeof(paramOne);
```
Comment (jaygraygray) [inline students/nicole/lab/drills/functions.js]: Did you test this code to see if it works?

`typeof` isn't a function, it's a keyword, so you don't invoke it with `()` You use it like this: `typeof variableName`

And

File: students/nicole/lab/drills/functions.js [hunk 3]
```diff
@@ -69,17 +69,62 @@
...
+
 // Declare three variables: firstName, middleName, lastName
 // Create a function called generateFullName that accepts three parameters
 // The function should return a concatenated string, creating a full
-// name from the three passed in
\ No newline at end of file
+// name from the three passed in
+
+
+function generateFullName(paraOne, paraTwo, paraThree){
```
Comment (jaygraygray) [inline students/nicole/lab/drills/functions.js:126]: This function would only ever return your full name. You're not using the `paraOne, paraTwo, paraThree` parameters in your function, instead you're declaring three new variables and ignoring the parameters.

Comment (ngray122) [reply students/nicole/lab/drills/functions.js:126]: I think I corrected it

**Comment by jaygraygray** [comment]
> I ran this code in repl and this returned "number" to the console

Oh shit, I didn't know that typeof is also a function! I always thought it was just used as a keyword. I stand corrected -- your way is totally correct to (but don't worry about changing your PR)

**Pull request merged**